### PR TITLE
[docs] fix autograd description on convex function case

### PR DESCRIPTION
**Pull request by dw61**
The sub-gradient of minimum norm is the least steep descent direction.

```python
import torch

x = torch.tensor([-2, -1, 0, 1, 2.], requires_grad=True)
torch.relu(x).sum().backward()
print(x.grad) # tensor([0., 0., 0., 1., 1.])

y = torch.tensor([-2, -1, 0, 1, 2.], requires_grad=True)
torch.abs(y).sum().backward()
print(y.grad) # tensor([-1., -1.,  0.,  1.,  1.])
```

cc @lezcano according to git blame
cc @albanD @soulitzer according to autograd codeowners
(How can I request a reviewer? I don't have the button on the right)

### CODE REVIEW

File: docs/source/notes/autograd.rst [hunk 1]
```diff
@@ -97,7 +97,7 @@ Unfortunately many of the functions we use in practice do not have this property
 To try and reduce the impact of functions that are non-differentiable, we define the gradients of the elementary operations by applying the following rules in order:
 
 #. If the function is differentiable and thus a gradient exists at the current point, use it.
-#. If the function is convex (at least locally), use the sub-gradient of minimum norm (it is the steepest descent direction).
+#. If the function is convex (at least locally), use the sub-gradient of minimum norm (it is the least steep descent direction).
```
Comment (lezcano) [inline docs/source/notes/autograd.rst:100]:
```suggestion
#. If the function is convex (at least locally), use the sub-gradient of minimum norm (it is the steepest ascent direction).
```

**Review by lezcano — APPROVED** [review]
Let's remove the double negation, but  yeah, thank you for catching this!

File: docs/source/notes/autograd.rst (hunk 1 above)
Comment (dw61) [reply docs/source/notes/autograd.rst]: It's actually not the steepest ascent. For example, in y = abs(x) at x=0, the steepest descent / ascent sub gradient is -1  and 1 interchangeably, but pytorch uses gradient at 0. see the PR description https://github.com/pytorch/pytorch/pull/148658#issue-2900027217. I changed it to `flattest descent direction` to avoid double negation @lezcano

Comment (dw61) [reply docs/source/notes/autograd.rst]: done! cc @albanD for review
